src/dnx/servicing.cpp

```diff
@@ -70,8 +70,21 @@ namespace dnx
                 ? utils::path_combine(servicing_root_parent, std::wstring(L"Microsoft DNX\\Servicing"))
                 : servicing_root_parent;
 
+            if (is_default_servicing_location && !utils::directory_exists(servicing_root))
+            {
+                trace_writer.write(L"The default servicing root does not exist.", true);
+                return std::wstring{};
+            }
+
             auto servicing_manifest_path = utils::path_combine(servicing_root, std::wstring(L"index.txt"));
 
+            if (!utils::file_exists(servicing_manifest_path))
+            {
+                throw std::runtime_error(std::string("The servicing index file at: ")
+                    .append(dnx::utils::to_string(servicing_manifest_path))
```

Comment:
@blowdart - is it OK to return the path in the exception? (Note that I am doing a similar change in Helios and there it could potentially show on the error page)